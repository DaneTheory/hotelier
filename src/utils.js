import isElectron from 'is-electron';

export default class Utils {
  static openExternalLink = (url) => {
    if (isElectron()) {
      window.shell.openExternal(url);
    } else {
      window.open(url);
    }
  }
}
