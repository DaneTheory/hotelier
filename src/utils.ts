import isElectron from 'is-electron';

export default class Utils {
  static isElectron = () => {
    return isElectron();
  }

  static openExternalLink = (url: string) => {
    if (Utils.isElectron()) {
      window.shell.openExternal(url);
    } else {
      window.open(url);
    }
  }
}
