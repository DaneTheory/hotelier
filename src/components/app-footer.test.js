import React from 'react';
import AppFooter from './app-footer';
import utils from '../utils';
import { shallow } from 'enzyme';

describe('AppFooter', () => {
  function create() {
    return shallow(<AppFooter/>);
  }

  it('matches the snapshot', () => {
    const rendered = create();
    expect(rendered).toMatchSnapshot();
  });

  it('can open hotel link', () => {
    const instance = create().instance();
    jest.spyOn(utils, 'openExternalLink');
    instance.openHotel();
    expect(utils.openExternalLink).toHaveBeenCalledWith('http://localhost:2000');
  });

  it('can close the app', () => {
    jest.spyOn(window, 'close');
    create().find('#close-app').simulate('click');
    expect(window.close).toHaveBeenCalled();
  });
});
