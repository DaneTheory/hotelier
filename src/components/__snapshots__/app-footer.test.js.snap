// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppFooter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="toolbar-actions pull-left"
>
          <button
                    className="btn btn-default"
                    onClick={[Function]}
          >
                    Hotel
          </button>
</div>,
        <div
          className="toolbar-actions pull-right"
>
          <div
                    className="btn-group"
          >
                    <button
                              className="btn btn-default"
                              onClick={undefined}
                    >
                              <span
                                        className="icon icon-arrows-ccw"
                                        title="Refresh"
                              />
                    </button>
                    <button
                              className="btn btn-default"
                              id="close-app"
                              onClick={[Function]}
                    >
                              <span
                                        className="icon icon-cancel"
                                        title="Quit"
                              />
                    </button>
          </div>
</div>,
      ],
      "className": "toolbar toolbar-footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-default"
            onClick={[Function]}
>
            Hotel
</button>,
          "className": "toolbar-actions pull-left",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hotel",
            "className": "btn btn-default",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Hotel",
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="btn-group"
>
            <button
                        className="btn btn-default"
                        onClick={undefined}
            >
                        <span
                                    className="icon icon-arrows-ccw"
                                    title="Refresh"
                        />
            </button>
            <button
                        className="btn btn-default"
                        id="close-app"
                        onClick={[Function]}
            >
                        <span
                                    className="icon icon-cancel"
                                    title="Quit"
                        />
            </button>
</div>,
          "className": "toolbar-actions pull-right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn btn-default"
                onClick={undefined}
>
                <span
                                className="icon icon-arrows-ccw"
                                title="Refresh"
                />
</button>,
              <button
                className="btn btn-default"
                id="close-app"
                onClick={[Function]}
>
                <span
                                className="icon icon-cancel"
                                title="Quit"
                />
</button>,
            ],
            "className": "btn-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="icon icon-arrows-ccw"
                  title="Refresh"
/>,
                "className": "btn btn-default",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon icon-arrows-ccw",
                  "title": "Refresh",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="icon icon-cancel"
                  title="Quit"
/>,
                "className": "btn btn-default",
                "id": "close-app",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon icon-cancel",
                  "title": "Quit",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="toolbar-actions pull-left"
>
            <button
                        className="btn btn-default"
                        onClick={[Function]}
            >
                        Hotel
            </button>
</div>,
          <div
            className="toolbar-actions pull-right"
>
            <div
                        className="btn-group"
            >
                        <button
                                    className="btn btn-default"
                                    onClick={undefined}
                        >
                                    <span
                                                className="icon icon-arrows-ccw"
                                                title="Refresh"
                                    />
                        </button>
                        <button
                                    className="btn btn-default"
                                    id="close-app"
                                    onClick={[Function]}
                        >
                                    <span
                                                className="icon icon-cancel"
                                                title="Quit"
                                    />
                        </button>
            </div>
</div>,
        ],
        "className": "toolbar toolbar-footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-default"
              onClick={[Function]}
>
              Hotel
</button>,
            "className": "toolbar-actions pull-left",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hotel",
              "className": "btn btn-default",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Hotel",
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="btn-group"
>
              <button
                            className="btn btn-default"
                            onClick={undefined}
              >
                            <span
                                          className="icon icon-arrows-ccw"
                                          title="Refresh"
                            />
              </button>
              <button
                            className="btn btn-default"
                            id="close-app"
                            onClick={[Function]}
              >
                            <span
                                          className="icon icon-cancel"
                                          title="Quit"
                            />
              </button>
</div>,
            "className": "toolbar-actions pull-right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn btn-default"
                  onClick={undefined}
>
                  <span
                                    className="icon icon-arrows-ccw"
                                    title="Refresh"
                  />
</button>,
                <button
                  className="btn btn-default"
                  id="close-app"
                  onClick={[Function]}
>
                  <span
                                    className="icon icon-cancel"
                                    title="Quit"
                  />
</button>,
              ],
              "className": "btn-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="icon icon-arrows-ccw"
                    title="Refresh"
/>,
                  "className": "btn btn-default",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon icon-arrows-ccw",
                    "title": "Refresh",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="icon icon-cancel"
                    title="Quit"
/>,
                  "className": "btn btn-default",
                  "id": "close-app",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon icon-cancel",
                    "title": "Quit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
